field/gf251/element_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2025 Consensys Software Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by go-corset DO NOT EDIT
+
+package gf251
+
+import (
+	"math/big"
+	"math/rand/v2"
+	"testing"
+
+	"github.com/consensys/go-corset/pkg/util/assert"
+)
+
+func TestMul(t *testing.T) {
+	var i, j, m big.Int
+
+	m.SetUint64(modulus)
+
+	for range 10000 {
```

Comment:
I would suggest a different testing strategy in this case.  It should be possible to enumerate **all combinations**, and I would do this.